checks/evaluation/branch_protection_test.go

```diff
@@ -18,563 +18,1691 @@ import (
+					Outcome: finding.OutcomeNegative,
+					Values: map[string]int{
+						"": 1,
+					},
+				},
+				{
+					Probe:   "dismissesStaleReviews",
+					Outcome: finding.OutcomeNegative,
+					Values: map[string]int{
+						"": 1,
+					},
+				},
+				{
+					Probe:   "requiresApproversForPullRequests",
+					Outcome: finding.OutcomeNegative,
+					Values: map[string]int{
+						"":                          1,
+						"numberOfRequiredReviewers": 0,
+					},
+				},
+				{
+					Probe:   "requiresCodeOwnersReview",
+					Outcome: finding.OutcomeNegative,
+					Values: map[string]int{
+						"": 1,
+					},
+				},
+				{
+					Probe:   "requiresLastPushApproval",
+					Outcome: finding.OutcomeNegative,
+					Values: map[string]int{
+						"": 1,
+					},
+				},
+				{
+					Probe:   "requiresUpToDateBranches",
+					Outcome: finding.OutcomePositive,
+					Values: map[string]int{
+						"": 1,
+					},
+				},
+				{
+					Probe:   "runsStatusChecksBeforeMerging",
+					Outcome: finding.OutcomePositive,
+					Values: map[string]int{
+						"": 1,
+					},
+				},
+				{
+					Probe:   "requiresPRsToChangeCode",
+					Outcome: finding.OutcomePositive,
+					Values: map[string]int{
+						"": 1,
 					},
 				},
 			},
-		},
-		{
-			name: "Nothing is enabled and values are nil",
-			expected: scut.TestReturn{
-				Error:         nil,
-				Score:         0,
-				NumberOfWarn:  3,
-				NumberOfInfo:  0,
-				NumberOfDebug: 4,
-			},
-			branch: &clients.BranchRef{
-				Name:      &branchVal,
-				Protected: &trueVal,
+			result: scut.TestReturn{
+				Error: sce.ErrScorecardInternal,
+				Score: checker.InconclusiveResultScore,
 			},
 		},
 		{
 			name: "Required status check enabled",
-			expected: scut.TestReturn{
-				Error:         nil,
-				Score:         4,
-				NumberOfWarn:  5,
-				NumberOfInfo:  5,
-				NumberOfDebug: 0,
-			},
-			branch: &clients.BranchRef{
-				Name:      &branchVal,
-				Protected: &trueVal,
-				BranchProtectionRule: clients.BranchProtectionRule{
-					RequiredPullRequestReviews: clients.PullRequestReviewRule{
-						Required:                     &trueVal,
-						DismissStaleReviews:          &falseVal,
-						RequireCodeOwnerReviews:      &falseVal,
-						RequiredApprovingReviewCount: &zeroVal,
-					},
-					CheckRules: clients.StatusChecksRule{
-						RequiresStatusChecks: &trueVal,
-						UpToDateBeforeMerge:  &trueVal,
-						Contexts:             []string{"foo"},
-					},
-					EnforceAdmins:           &falseVal,
-					RequireLastPushApproval: &falseVal,
-					RequireLinearHistory:    &falseVal,
-					AllowForcePushes:        &falseVal,
-					AllowDeletions:          &falseVal,
+			findings: []finding.Finding{
+				{
+					Probe:   "blocksDeleteOnBranches",
+					Outcome: finding.OutcomePositive,
+					Values: map[string]int{
+						"main": 1,
+					},
+				},
+				{
+					Probe:   "blocksForcePushOnBranches",
+					Outcome: finding.OutcomePositive,
+					Values: map[string]int{
+						"main": 1,
+					},
```

Comment:
optional and/or followup ( I can pick this up if you want):

can we just create a pre-defined var (for the probes that don't have other values, like the number of reviewers) so we can cut down on the lines and increase readability?

```
mainBranchValues := map[string]int{
	"main": 1,
},
```

And then the tests can just reference: 
```
Probe:   "blocksForcePushOnBranches",
Outcome: finding.OutcomePositive,
Values: mainBranchValues
```

Or even just a helper func to make a finding based on name + outcome?
```go
func testFinding(probe string, outcome finding.Outcome) finding.Outcome {
	return finding.Finding{
		Probe: probe,
		Outcome: outcome,
		Values: map[string]int{
			"main": 1,
		},
	}
}
```

And then we cut down on the toil/lines?
```go
findings: []finding.Finding{
	testFinding("blocksDeleteOnBranches", finding.OutcomePositive),
	testFinding("blocksForcePushOnBranches", finding.OutcomePositive),
	// and so on
```

Reply:
> ( I can pick this up if you want)

Thanks, that would be great.

Reply:
#3904 3904